src/examples/TDXContainers.sol

```diff
@@ -0,0 +1,189 @@
+pragma solidity ^0.8.13;
+
+import "../crypto/secp256k1.sol";
+import "../crypto/encryption.sol";
+import "../KeyManager.sol";
+
+struct ContainerConfig {
+    uint version;
+    /* type? */
+    bytes32 imageHash;
+    string[] containerArgs;
+    /* container features */
+    /* vm features */
+    /* hardware stuff */
+    /* maybe required measurement */
+}
+
+contract TDXContainers {
+    address owner;
+    KeyManager_v0 keymgr;
+
+    /* TODO: instead require measurement is one of expected */
+    bytes32 quoteMeasurementHash;
+    bytes hostPubkey; // Passed in with quote
+
+    /* TODO (maybe): verify certificate onchain against a root */
+    string public pckCertPem;
+    string public pckCrlPem;
+    string public tcbInfo;
+    string public qeIdentity;
+
+    /* Maps keccak256(ContainerConfig) -> ContainerConfig */
+    mapping(bytes32 => ContainerConfig) public pendingContainers;
+    mapping(bytes32 => ContainerConfig) public approvedContainers;
+
+    /* Maps keccak256(keccak256(ContainerConfig), msg.sender) -> ContainerConfig */
+    mapping(bytes32 => ContainerConfig) public runningContainers;
+
+    constructor(KeyManager_v0 _keymgr, string memory _pckCertPem, string memory _pckCrlPem, string memory _tcbInfo, string memory _qeIdentity) {
+        owner = msg.sender;
+        keymgr = _keymgr;
+
+        /* TODO: should we allow more than one pck? */
+        pckCertPem = _pckCertPem;
+        pckCrlPem = _pckCrlPem;
+
+        /* TODO: allow more than one tcbInfo */
```

Comment:
Dealing with TCBInfos really needs its own entire separate on-chain service. They should be validated, and we can decide whether to honor CRL or expiries etc.